### PR TITLE
988 create log builder to log common monitoring data such as user ids

### PR DESCRIPTION
**Pull request by sadatmalik**
Note that this PR has automatically optimised imports on about 40 typescript files in accordance with our coding styles. Most of these optimised imports are in unit test "spec" files.

Aside from this, the PR file changes include 56 Java classes, introducing the following updates:

- Adds a LogBuilder in the new logging package for standardised log message construction
- Updates all classes to use the new LogBuilder, replacing all default logging
- Uses @Slf4j Lombok annotation throughout the project for consistency
- Introduces compact JSON strings for elastic search query logging
- Resolves a few IntelliJ warnings in SavedSearchServiceImpl and CandidateServiceImpl

### CODE REVIEW

File: server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java [hunk 1]
```diff
@@ -996,7 +1282,11 @@ public String migrateSurvey() {
 
             Connection targetConn = DriverManager.getConnection(targetJdbcUrl, targetUser, targetPwd);
 
-            log.info("Migration survey data for candidates");
+            LogBuilder.builder(log)
+                .user(authService.getLoggedInUser())
+                .action("MigrateSurvey")
+                .message("Migration survey data for candidates")
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java:1288]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We found 15 occurrences of messages that are similar to the selected lines. However, the message in the selected lines is not consistent with the other messages. We recommend you keep the messages in your code consistent. The following are some examples of similar messages in your code: 
talentcatalog/server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java#L1840:             .message(&#34;Migration data for candidate educations&#34;)

talentcatalog/server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java#L1910:             .message(&#34;Migration data for candidate occupations&#34;)

File: server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java [hunk 2]
```diff
@@ -1042,17 +1332,35 @@
...
+            LogBuilder.builder(log)
+                .user(authService.getLoggedInUser())
+                .action("MigrateSurvey")
+                .message("candidates - saving batch " + count)
+                .logInfo();
+
+            LogBuilder.builder(log)
+                .user(authService.getLoggedInUser())
+                .action("MigrateSurvey")
+                .message("Migration survey data for candidates")
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java:1355]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We found 15 occurrences of messages that are similar to the selected lines. However, the message in the selected lines is not consistent with the other messages. We recommend you keep the messages in your code consistent. The following are some examples of similar messages in your code: 
talentcatalog/server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java#L1840:             .message(&#34;Migration data for candidate educations&#34;)

talentcatalog/server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java#L1910:             .message(&#34;Migration data for candidate occupations&#34;)

File: server/src/main/java/org/tctalent/server/service/db/es/ElasticsearchServiceImpl.java [hunk 3]
```diff
@@ -164,7 +181,11 @@ private BoolQueryBuilder computeFindByNumberQuery(String number) {
     boolQuery = filterOnDeletedStatus(boolQuery);
     boolQuery = filterOnSourceCountryRestrictions(boolQuery);
 
-    log.debug("Elasticsearch query:\n" + boolQuery);
+    LogBuilder.builder(log)
+        .action("ElasticsearchServiceImpl.computeFindByNumberQuery")
+        .message("Constructed Elasticsearch query:\n " + boolQuery)
+        .logDebug();
+
     return boolQuery;
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/service/db/es/ElasticsearchServiceImpl.java:189]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 173:189, 223:239.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.

File: server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java [hunk 4]
```diff
@@ -1846,4 +1917,24 @@ private BoolQueryBuilder processElasticRequest(SearchCandidateRequest searchRequ
         return boolQueryBuilder;
     }
 
+    private String convertToJson(BoolQueryBuilder boolQueryBuilder) {
+        try {
+            XContentBuilder builder = XContentFactory.jsonBuilder();
```
Comment (sadatmalik) [inline server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *builder*. The resource is referenced at lines 1923, 1925. The resource is closed at line 1924. There are other execution paths that do not contain closure statements, for example, when BoolQueryBuilder.toXContent throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

File: server/src/main/java/org/tctalent/server/service/db/es/ElasticsearchServiceImpl.java (hunk 3 above)
Comment (sadatmalik) [reply server/src/main/java/org/tctalent/server/service/db/es/ElasticsearchServiceImpl.java:189]: Choosing not to make this refactoring in order tol keep the usage of LogBuilder consistent throughout the project.

File: server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java (hunk 4 above)
Comment (sadatmalik) [reply server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java]: Good catch. I have pushed an update to use try with resources to address this potential issue.

**Review by camerojo — DISMISSED** [review]
Approved

File: server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java (hunk 2 above)
Comment (sadatmalik) [reply server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java:1355]: These are log output messages for different scenarios; therefore they are necessarily different.

File: server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java (hunk 1 above)
Comment (sadatmalik) [reply server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java:1288]: These are logging messages for different application scenarios; therefore, they are different.

**Review by camerojo — APPROVED** [review]

**Pull request merged**